### ISSUE TITLE
Processing a new waterway relation 'misses' wipes of individual members

### ISSUE THREAD

**Issue by skylerbunny** [openstreetmap/Nominatim#848]
I think where this is done is here:

https://github.com/openstreetmap/Nominatim/blob/c44324fda5ea057dfde87ddee2a749448cf9b146/sql/functions.sql#L1349

Example: http://www.openstreetmap.org/relation/7712135

Despite the fact that version one (and any subsequent version, although I suspect v1 is the only relevant one) of this relation, as created in http://www.openstreetmap.org/changeset/53545490 contains ways 337489117, 288263255, 321177814, and 321217190, all four appear in Nominatim search results.

Expected: Only http://www.openstreetmap.org/relation/7712135 should appear in the result, since the above listed four ways are members of the relation. I think - THINK this is because waterway=river does not match waterway=stream in the relation members, but it is common practice to make a waterway's individual way vary depending upon the width of that waterway, and to assign the 'widest' to the relation.

**Comment by skylerbunny** [comment]
tl;dr If I'm right, it's probably sufficient to check a member way as 'Does this way contain waterway=*'? or maybe 'waterway=[stream|river|whatever]'? Rather than waterway='relation parent exactly'.

**Comment by lonvia** [comment]
Oddly enough, the last suggestion in #155 proposed to accept other types of waterway but it looks like I never implemented that in the end. I wonder why.
